AI Features: add AI Skills topic (#8830)
DevExpress/devextreme-documentation: concepts/90 AI Features/30 DevExpress AI Skills.md

> DevExpress AI Skills is a public repository of reusable agent skills for DevExpress products, including DevExtreme. The DevExtreme skills are located in the [skills](https://github.com/DevExpress/agent-skills/tree/main/plugins/dx-devextreme/skills/skills) folder of the [DevExpress/agent-skills](https://github.com/DevExpress/agent-skills) repository and are designed for AI coding assistants such as GitHub Copilot, Claude Code, Cursor, and JetBrains AI Assistant.

Reviewer: ```suggestion
DevExpress AI Skills is a public repository of reusable agent skills for DevExpress products (including DevExtreme). DevExtreme-specific skills are located in the [skills](https://github.com/DevExpress/agent-skills/tree/main/plugins/dx-devextreme/skills/skills) folder of the [DevExpress/agent-skills](https://github.com/DevExpress/agent-skills) repository and are designed for AI coding assistants, such as GitHub Copilot, Claude Code, Cursor, and JetBrains AI Assistant.
```

While I'm reading this one...

> such as GitHub Copilot, Claude Code, Cursor, and JetBrains AI Assistant

I can't help but wonder if this a full list of supported AI coding assistants. Because if that's the case, perhaps we need to tweak this particular sentence a bit. If this is an incomplete/open-ended list, perhaps we also should consider certain optimizations. At the same time, we can leave the current version (as of I've made in this particular comment/suggestion).

I do hope this helps/makes sense to you.

> Use these skills when you want an AI assistant to follow DevExpress-specific patterns, choose the right APIs, and generate code that matches the product and framework you are working with.

Reviewer: ```suggestion
Use these skills when you want an AI assistant to follow DevExpress-specific patterns, use feature/task-specific APIs, and generate code that corresponds to the product and framework you are working with.
```

> Use these skills when you want an AI assistant to follow DevExpress-specific patterns, use feature/task-specific APIs, and generate code that corresponds to the product and framework you are working with.

Reviewer 2: ```suggestion
Use these skills when you want an AI assistant to follow DevExpress-specific patterns, use feature/task-specific APIs, and generate code that corresponds to the product and framework with which you are working.
```

> ## What Each Skill Contains

Reviewer: This section looks more than good to me. I do commend your efforts here!..

- Product-specific guidance for common DevExtreme scenarios
- Correct API names, import paths, and configuration patterns
- Ready-to-run examples for common scenarios
- Reusable instructions you can copy into a project-level or global skills folder

## DevExtreme Skills

> The repository includes skills for the following DevExtreme areas:

Reviewer: I cannot say I'm sure what do you mean by "areas" in this content and scope...

> The repository includes skills for the following DevExtreme **areas**:

Any comments would've been helpful in this regard... At least the major part of the list are products/components (Data Grid, Scheduler, Chat, Button, Check Box, etc.).

> The repository includes skills for the following DevExtreme components and features:

Reviewer 2: ```suggestion
This repository includes skills for the following DevExtreme components and features:
```

- [CheckBox](/concepts/05%20UI%20Components/CheckBox/00%20Getting%20Started%20with%20CheckBox/00%20Getting%20Started%20with%20CheckBox.md '/Documentation/Guide/UI_Components/CheckBox/Getting_Started_with_CheckBox/')
- [NumberBox](/concepts/05%20UI%20Components/NumberBox/00%20Getting%20Started%20with%20NumberBox/00%20Getting%20Started%20with%20NumberBox.md '/Documentation/Guide/UI_Components/NumberBox/Getting_Started_with_NumberBox/')
- [TextBox](/concepts/05%20UI%20Components/TextBox/02%20Overview/00%20Overview.md '/Documentation/Guide/UI_Components/TextBox/Overview/')
- [TextArea](/concepts/05%20UI%20Components/TextArea/00%20Getting%20Started%20with%20TextArea/00%20Getting%20Started%20with%20TextArea.md '/Documentation/Guide/UI_Components/TextArea/Getting_Started_with_TextArea/')
- [DataSource](/concepts/70%20Data%20Binding/5%20Data%20Layer/0%20Overview.md '/Documentation/Guide/Data_Binding/Data_Layer/')
- [Theming](/concepts/60%20Themes%20and%20Styles/05%20Predefined%20Themes/00%20Predefined%20Themes.md '/Documentation/Guide/Themes_and_Styles/Predefined_Themes/')

> ## How To Use The Repository

Reviewer: Let's take a look...
> How **To** Use **The** Repository

I tend to agree with Copilot -- are you sure this is the exact capitalization you need?

```suggestion
## How to Use the Repository
```

P.S. It's kinda strange to see here...

1. Open the [skills](https://github.com/DevExpress/agent-skills/tree/main/plugins/dx-devextreme/skills/skills) folder in the DevExpress/agent-skills repository.

> 2. Copy the skill folders you need to your AI assistant's skills directory.

Reviewer: A little adjustment is in order (if I may):

```suggestion
2. Copy all skill folders you need to your AI assistant's skills directory.
```

3. Start a chat in agent mode and ask a product-specific question.

For agent-specific and IDE-specific setup instructions, refer to the [repository README](https://github.com/DevExpress/agent-skills/blob/main/README.md).

> ## When To Use AI Skills And When To Use MCP

Reviewer: Likewise, I find it strange to see every word capitalized...

```suggestion
## When to Use AI Skills and When to Use MCP
```

P.S. I would've stuck with common rules we've agreed on...

> For best results, use DevExpress AI Skills together with the [DevExpress MCP server](/concepts/90%20AI%20Features/20%20DevExpress%20MCP%20Server%20Configuration/00%20DevExpress%20MCP%20Server%20Configuration.md '/Documentation/Guide/AI_Features/DevExpress_MCP_Server_Configuration/'). Skills supply curated task patterns and product-specific rules, while the MCP server adds live documentation lookup and version-sensitive details. The two complement each other.

Reviewer: > Skills supply curated task patterns and product-specific rules, **while** the MCP server adds live documentation lookup

No comma is necessary before "while". However, this is not the only thing I can say regarding this paragraph:

```suggestion
For best results, use DevExpress AI Skills together with the [DevExpress MCP server](/concepts/90%20AI%20Features/20%20DevExpress%20MCP%20Server%20Configuration/00%20DevExpress%20MCP%20Server%20Configuration.md '/Documentation/Guide/AI_Features/DevExpress_MCP_Server_Configuration/'). Skills supply curated task patterns and product-specific rules while the MCP server adds live documentation lookup and version-sensitive details. DevExpress AI Skills and MCP server complement each other.
```

> If you use only one:

Reviewer: This introductory line looks strange together with the following two items. First of all, this introduction does not convey the scenario we're dealing with upfront. From the other hand, it repeats use-use when taken together with the items listed below...

What did you want to say here, I wonder? Did you wanted to list the scenarios when one may want to use either DevExpress AI Skills or DevExpress MCP Server independently from each other?

I'll make a honest attempt to rewrite this entire section, however, I'm unsure if the result can serve as the final version... Alright. Let's get over with:

```suggestion
While DevExpress AI Skills and DevExpress MCP Server produce the best results when used together, separate usage of these tools may be beneficial in the following scenarios:
```

> For best results, use DevExpress AI Skills together with the [DevExpress MCP server](/concepts/90%20AI%20Features/20%20DevExpress%20MCP%20Server%20Configuration/00%20DevExpress%20MCP%20Server%20Configuration.md '/Documentation/Guide/AI_Features/DevExpress_MCP_Server_Configuration/'). Skills supply curated task patterns and product-specific rules while the MCP server adds live documentation lookup and version-sensitive details. DevExpress AI Skills and MCP server complement each other.

Reviewer 2: ```suggestion
For best results, use DevExpress AI Skills together with the [DevExpress MCP Server](/concepts/90%20AI%20Features/20%20DevExpress%20MCP%20Server%20Configuration/00%20DevExpress%20MCP%20Server%20Configuration.md '/Documentation/Guide/AI_Features/DevExpress_MCP_Server_Configuration/'). Skills supply curated task patterns and product-specific rules while the MCP Server adds live documentation lookup and version-sensitive details. DevExpress AI Skills and MCP Server complement each other.
```

> - Use DevExpress AI Skills when you want the assistant to follow curated task patterns and product-specific rules without a network connection to documentation.

Reviewer: Like this:

```suggestion
- When you need the assistant to follow curated task patterns and product-specific rules without a network connection to documentation, use DevExpress AI Skills.
```

> - Use the DevExpress MCP server when you need live documentation lookup, version-sensitive details, or a direct connection to the DevExpress documentation library.

Reviewer: ```suggestion
- When you need live documentation lookup, version-sensitive details, or a direct connection to the DevExpress documentation library, use the DevExpress MCP server.
```

> [note] We recommend using a top-tier model for the most accurate results. Lower-tier models may ignore skill instructions or produce less reliable output.

Reviewer: I've never seen this syntax:

> [note]

However, I do believe you know what you are doing... As for the wording in this paragraph, everything looks good in my opinion, and I do commend you for your efforts.

> - When you need live documentation lookup, version-sensitive details, or a direct connection to the DevExpress documentation library, use the DevExpress MCP server.

Reviewer 2: ```suggestion
- When you need live documentation lookup, version-sensitive details, or a direct connection to the DevExpress documentation library, use the DevExpress MCP Server.
```
